modules/skybrowser/src/screenspaceskybrowser.cpp

```diff
@@ -0,0 +1,276 @@
+ * Copyright (c) 2014-2022                                                               *
+ *                                                                                       *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of this  *
+ * software and associated documentation files (the "Software"), to deal in the Software *
+ * without restriction, including without limitation the rights to use, copy, modify,    *
+ * merge, publish, distribute, sublicense, and/or sell copies of the Software, and to    *
+ * permit persons to whom the Software is furnished to do so, subject to the following   *
+ * conditions:                                                                           *
+ *                                                                                       *
+ * The above copyright notice and this permission notice shall be included in all copies *
+ * or substantial portions of the Software.                                              *
+ *                                                                                       *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,   *
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A         *
+ * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT    *
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF  *
+ * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE  *
+ * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                                         *
+ ****************************************************************************************/
+#include <modules/skybrowser/include/screenspaceskybrowser.h>
+
+#include <modules/skybrowser/skybrowsermodule.h>
+#include <modules/skybrowser/include/utility.h>
+#include <openspace/engine/globals.h>
+#include <openspace/engine/windowdelegate.h>
+#include <openspace/engine/moduleengine.h>
+#include <openspace/rendering/renderengine.h>
+#include <ghoul/logging/logmanager.h>
+#include <ghoul/misc/dictionaryjsonformatter.h> 
+#include <ghoul/opengl/texture.h>
+#include <optional>
+#include <glm/gtx/color_space.hpp> 
+#include <random>
+
+namespace {
+    constexpr const char* _loggerCat = "ScreenSpaceSkyBrowser";
+
+    constexpr const openspace::properties::Property::PropertyInfo AnimationSpeedInfo =
+    {
+        "AnimationSpeed",
+        "Field Of View Animation Speed",
+        "The factor which is multiplied with the animation of the field of view."
+    };
+    constexpr const openspace::properties::Property::PropertyInfo TextureQualityInfo =
+    {
+        "TextureQuality",
+        "Quality of Texture",
+        "A parameter to set the resolution of the texture. 1 is full resolution and "
+        "slower framerate. Lower value means lower resolution of texture and faster "
+        "frame rate."
```

Comment:
Nitpicky, but two different ways of writing "framerate" in the same text. Don't know which one to prefer, but be consistent